### PR TITLE
Don't use expect script for slurm-gasnetrun_ibv launcher

### PR DESCRIPTION
**Pull request by ronawho**
This is similar to https://github.com/chapel-lang/chapel/pull/950

There is no reason to be using an expect script, it was only used because the
launcher was ported from pbs-gasnetrun_ibv, which maybe required it. The expect
script added carriage returns, which meant you couldn't just use it under
start_test. It also obviously required expect, which is not available on all
systems (e.g. it's not available on internal Cray CS machines.)

Additionally the expect script merged stderr into stdout, so this is a
prerequisite for https://github.com/chapel-lang/chapel/issues/12259

This also quotes the env var propagation, which caused problems on systems
where there were env vars with special characters.

Testing (hellos+types/string/dmk/utf8_env): 
 - [x] chapcs w/ and w/o launchcmd 
 - [x] internal Cray CS w/ and w/o launchmd

### CODE REVIEW

**Comment by ronawho** [comment]
This is easiest to review commit-by-commit ignoring whitespace.

File: runtime/src/launch/slurm-gasnetrun_ibv/launch-slurm-gasnetrun_ibv.c [hunk 1]
```diff
@@ -305,10 +305,7 @@ static char* chpl_launch_create_command(int argc, char* argv[],
 static void chpl_launch_cleanup(void) {
   if (!debug) {
     char command[2*FILENAME_MAX];
-    if (getenv("CHPL_LAUNCHER_USE_SBATCH") == NULL) {
-      sprintf(command, "rm %s", expectFilename);
-      system(command);
-    } else {
+    if (getenv("CHPL_LAUNCHER_USE_SBATCH") != NULL) {
       sprintf(command, "rm %s", slurmFilename);
       system(command);
```
Comment (dmk42) [inline runtime/src/launch/slurm-gasnetrun_ibv/launch-slurm-gasnetrun_ibv.c]: This isn't your doing, but this seems fragile.  Can we use `remove(slurmFilename)` instead?

Comment (ronawho) [reply runtime/src/launch/slurm-gasnetrun_ibv/launch-slurm-gasnetrun_ibv.c]: We use `unlink(slurmFilename)` in the slurm-srun launcher. Ok with switching to that or is there some reason to prefer `remove()`?

Comment (dmk42) [reply runtime/src/launch/slurm-gasnetrun_ibv/launch-slurm-gasnetrun_ibv.c]: That's fine.  It's probably best to do whichever we do elsewhere.  `remove()` is from the C standard and `unlink()` is from the POSIX standard.

**Pull request merged**